src/mpid/common/thread/mpidu_thread_fallback.h

```diff
@@ -113,29 +123,36 @@ M*/
             per_thread->lock_depth++;                                   \
         }                                                               \
     } while (0)
+
+#if MPICH_THREAD_GRANULARITY == MPICH_THREAD_GRANULARITY__GLOBAL
+
+#define MPIDUI_THREAD_CS_ENTER_GLOBAL    MPIDUI_THREAD_CS_ENTER_REC
 #define MPIDUI_THREAD_CS_ENTER_POBJ(mutex) do {} while (0)
+#define MPIDUI_THREAD_CS_ENTER_VNI_GLOBAL MPIDUI_THREAD_CS_ENTER_GLOBAL
```

Comment:
@hajimefu , you can use the `VNI` granularity with the `ALLFUNC` lock as long as you know it is created (which is the case here since we reuse the existing large critical section for many calls. So
Case 1: use `VNI_GLOBAL`
Case 2: Use `VNI`